accepted/single-file/extract.md

```diff
@@ -0,0 +1,81 @@
+# Extracting Bundled Files to Disk
+
+When running the single-file, the host will need to extract some bundled files resources out to the disk before loading them. This is particularly true for native DLLs because of operating system API restrictions -- for example, Windows doesn't support a `LoadLibrary()` API that reads in from a stream. 
+
+We now explore the options with respect to the location and life-time of the extracted files.
+
+## Time of Extraction
+
+* **Startup**: The host spills the necessary files at startup, and communicates the location to the runtime.
+* **Lazy:** The runtime provides callbacks to extract a bundled file to disk when it is actually necessary to be loaded. 
+
+Extraction at startup is easier to implement, as the runtime needs no knowledge of the extraction. 
+
+Lazy extraction saves startup cost, particularly if many ready-to-run images are bundled. The lazy extraction feature is likely useful only when files are extracted on every run.
+
+Mono extracts native libraries at startup. Costura extracts and explicitly loads the native-libraries at startup.
+
+## Temporary Extraction
+
+The first option is to extract out the necessary files to a temporary location, and clean them out on exit. The requirements for this path are:
+
+* Extract to a location within the temp-directory, so that if some files are not removed (ex: because an app crashed before cleanup), they are removed when the temp-directory is purged.
+* Be short, to reduce the risk of running over path length limit.
+* Be randomized, in order to support concurrent launches of the same app.
+
+The proposed extraction path is `%TEMP\.net\<random-code>` / `$TMPDIR/.net/<random-code>` . 
```

Comment:
Would this be configurable? I could imagine that some applications might rather have this be `%TEMP%\MyCompany\<random-code>`.

Reply:
The `.net` part just signifies that these are temporary files created by .net-core. This way, the entire `.net` folder can be easily cleaned up if necessary. We could, of course, have `%temp%\.net\app\...` etc. But we choose shorter file names -- so that we don;t hit the max-path limit.

Reply:
Yeah, I do understand the idea behind this but I could still see some situations where having a `.net` in the path would be unwanted. The fact that the application is a .NET application is an implementation detail after all.